protocol/app/ante/timestampnonce.go

```diff
@@ -0,0 +1,55 @@
+package ante
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	accountpluskeeper "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/keeper"
+)
+
+// TODO: combine increment sequence and sequence verification into one decorator
+// https://github.com/cosmos/cosmos-sdk/pull/18817
+type IncrementSequenceDecorator struct {
+	ak ante.AccountKeeper
+}
+
+func NewIncrementSequenceDecorator(ak ante.AccountKeeper) IncrementSequenceDecorator {
+	return IncrementSequenceDecorator{
+		ak: ak,
+	}
+}
+
+func (isd IncrementSequenceDecorator) AnteHandle(
+	ctx sdk.Context,
+	tx sdk.Tx,
+	simulate bool,
+	next sdk.AnteHandler,
+) (sdk.Context, error) {
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
+	if !ok {
+		return ctx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "invalid transaction type")
+	}
+
+	signatures, err := sigTx.GetSignaturesV2()
+	if err != nil {
+		return sdk.Context{}, err
+	}
+
+	for _, signature := range signatures {
```

Comment:
Note this changes the semantics of `IncrementSequenceDecorator`. It used to increment sequence for all `sigTx.GetSigners()`, now it's iterating over all the signatures (everyone who signed the transactions). These are not inherently the same set, so it's relying on upstream logic to call `SigVerifyDecorator` correctly.

Reply:
Is there a need to revert to iterating on signers and check that the signer has a signature?